java/libphonenumber/src/com/google/i18n/phonenumbers/MetadataManager.java

```diff
@@ -89,22 +85,54 @@ static CodedInputByteBufferNano convertStreamToByteBuffer(ObjectInputStream in,
     return CodedInputByteBufferNano.newInstance(outputStream.toByteArray());
   }
 
-  private static void loadAlternateFormatsMetadataFromFile(int countryCallingCode) {
-    InputStream source = PhoneNumberMatcher.class.getResourceAsStream(
-        ALTERNATE_FORMATS_FILE_PREFIX + "_" + countryCallingCode);
-    ObjectInputStream in = null;
+  /**
+   * Loads and returns the metadata protocol buffer from the given stream and closes the stream.
+   *
+   * @param source      the non-null stream from which metadata is to be read.
+   * @param bufferSize  the size of the buffer in bytes used to convert the stream to a
+                        nanoproto-compatible {@code CodedInputByteBufferNano}.
+   * @return            the loaded metadata protocol buffer.
+   */
+  static PhoneMetadataCollection loadMetadataAndCloseInput(InputStream source, int bufferSize) {
+    ObjectInputStream ois = null;
     try {
-      in = new ObjectInputStream(source);
-      CodedInputByteBufferNano byteBuffer = convertStreamToByteBuffer(in, BUFFER_SIZE);
-      PhoneMetadataCollection alternateFormats = new PhoneMetadataCollection();
-      alternateFormats.mergeFrom(byteBuffer);
-      for (PhoneMetadata metadata : alternateFormats.metadata) {
-        callingCodeToAlternateFormatsMap.put(metadata.countryCode, metadata);
+      try {
+        ois = new ObjectInputStream(source);
+      } catch (IOException e) {
+        throw new RuntimeException("cannot load/parse metadata", e);
+      }
+      PhoneMetadataCollection metadataCollection = new PhoneMetadataCollection();
+      try {
+        metadataCollection.mergeFrom(convertStreamToByteBuffer(ois, bufferSize));
+      } catch (IOException e) {
+        throw new RuntimeException("cannot load/parse metadata", e);
       }
-    } catch (IOException e) {
-      LOGGER.log(Level.WARNING, e.toString());
+      return metadataCollection;
     } finally {
-      close(in);
+      try {
+        if (ois != null) {
+          ois.close();
+        } else {
+          source.close();
```

Comment:
ois.close() also closes source. But if ois is null (as in if the ObjectInputStream initialization failed) then we want to attempt closing source independenly.